Sources/StreamChat/Controllers/ChannelController/ChannelController+SwiftUI.swift

```diff
@@ -19,7 +19,7 @@ extension _ChatChannelController {
         @Published public private(set) var channel: _ChatChannel<ExtraData>?
         
         /// The messages related to the channel.
-        @Published public private(set) var messages: [_ChatMessage<ExtraData>] = []
+        @Published public private(set) var messages: AnyRandomAccessCollection<_ChatMessage<ExtraData>> = .init([])
```

Comment:
We shouldn't expose this via `AnyRandomAccessCollection`, users then can't use `Int` for the index of the collection. Ideally, we'd expose the `LazyCachedMapCollection` as is, but we can't do that because it leaks the DTO type. So we probably need some kind of custom `AnyLazyCachedMapCollection<Element>: Collection` wrapper type for this, where the `Index == Int`.

Reply:
This comment is valid for all similar cases in the PR